src/operations/count.ts

```diff
@@ -53,6 +53,51 @@ class CountOperation extends CommandOperation {
   }
 }
 
+/**
+ * Build the count command.
+ *
+ * @function
+ * @param {Collection|Cursor} collectionOrCursor an instance of a collection or cursor
+ * @param {any} query The query for the count.
+ * @param {any} [options] Optional settings. See Collection.prototype.count and Cursor.prototype.count for a list of options.
+ */
+function buildCountCommand(collectionOrCursor: any, query: any, options?: any) {
```

Comment:
Should we utilize `Collection | Cursor` as the first arg's type here? We could add a helper/util that looks something like:
```ts
function isCursor(c: Collection | Cursor): c is Cursor { /* check if cursor */ }
function isCollection(c: Collection | Cursor): c is Collection { /* check if collection */ }
```
https://www.typescriptlang.org/docs/handbook/advanced-types.html#user-defined-type-guards

Reply:
yes, I think we should do that as part of [NODE-2732](https://jira.mongodb.org/browse/NODE-2732)